cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py

```diff
@@ -296,6 +299,48 @@ def create_self_config(ca, client):
     build_kubeconfig(server)
 
 
+@when('ceph-storage.available')
+def ceph_storage(ceph_admin):
+    '''A connection with ceph was created react to the ceph-storage relation
+    events here.'''
+    ceph_context = {
+        'mon_hosts': ceph_admin.mon_hosts(),
+        'fsid': ceph_admin.fsid(),
+        'auth_supported': ceph_admin.auth(),
+        'use_syslog': "true",
+        'ceph_public_network': '',
+        'ceph_cluster_network': '',
+        'loglevel': 1,
+        'hostname': socket.gethostname(),
+    }
+    # Install the ceph common utilities.
+    apt_install(['ceph-common'], fatal=True)
+
+    etc_ceph_directory = '/etc/ceph'
+    if not os.path.isdir(etc_ceph_directory):
+        os.makedirs(etc_ceph_directory)
+    charm_ceph_conf = os.path.join(etc_ceph_directory, 'ceph.conf')
+    # Render the ceph configuration from the ceph conf template
+    render('ceph.conf', charm_ceph_conf, ceph_context)
+
+    admin_key = os.path.join(etc_ceph_directory, 'ceph.client.admin.keyring')
+    try:
+        with open(admin_key, 'w') as key_file:
```

Comment:
Is it safe to always write this key to the file? Does this not fail idempotentcy rules?